server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -2311,6 +2329,20 @@ int getVectorDimensions() {
         ElementType getElementType() {
             return elementType;
         }
+
+        @Override
+        public BlockLoader blockLoader(MappedFieldType.BlockLoaderContext blContext) {
+            if (elementType != ElementType.FLOAT) {
+                throw new UnsupportedOperationException("Only float dense vectors are supported for now");
```

Comment:
We can work on this next, creating specific `BlockLoaders` for Byte and Bit field types.